src/txn/txn_ckpt.c

```diff
@@ -2012,7 +2014,8 @@ __checkpoint_lock_dirty_tree(
      * we want to periodically check if we need to delete old checkpoints that may have been in use
      * by an open cursor.
      */
-    if (!btree->modified && !force && is_checkpoint && is_wt_ckpt && !is_drop) {
+    if (!btree->modified && !force && is_checkpoint && is_wt_ckpt && !is_drop &&
+      !bm->can_truncate(btree->bm, session)) {
```

Comment:
As the comment suggests, it's a complicated test...
It does not need to be done in this PR but it would be great to split it and group things together that make sense.

Just reading it, I feel that we should update is_wt_ckpt to be !is_named_checkpoint

But like I said, it would be better in a different ticket.

Reply:
Is there somewhere we can document suggested changes to checkpoint or should I just create a ticket?

Reply:
We should create a ticket, it's tech debt.